src/datasets/arrow_dataset.py

```diff
@@ -3097,6 +3097,21 @@ def add_item(self, item: dict, new_fingerprint: str):
             fingerprint=new_fingerprint,
         )
 
+    def align_labels_with_mapping(self, label2id: Dict, label_column: str = "labels") -> "Dataset":
```

Comment:
I wouldn't set "labels" as the default and let the user specify explicitly which column they want to change.

Also could you add a docstring please ?
You can also add this method in the documentation in `main_classes.rst`

Reply:
done!